tests/test_core/test_new_alignment.py

```diff
@@ -2938,14 +2938,49 @@ def test_alignment_repr():
 @pytest.mark.parametrize("stop", range(6))
 @pytest.mark.parametrize("step", range(1, 3))
 @pytest.mark.parametrize("seqid", ("seq1", "seq2", "seq3", "seq4"))
-def test_alignment_slice(aligned_dict, start, stop, step, seqid):
+def test_alignment_slice_pos_step_gapped(aligned_dict, start, stop, step, seqid):
     """slicing an alignment should propogate the slice to aligned instances"""
     aln = new_alignment.make_aligned_seqs(aligned_dict, moltype="dna")
     sliced_aln = aln[start:stop:step]
     got = sliced_aln.seqs[seqid].gapped_seq
     expect = aligned_dict[seqid][start:stop:step]
     assert got == expect
 
+
+@pytest.mark.parametrize("start", range(6))
+@pytest.mark.parametrize("stop", range(6))
+@pytest.mark.parametrize("step", range(1, 3))
+@pytest.mark.parametrize("seqid", ("seq1", "seq2", "seq3", "seq4"))
+def test_alignment_slice_pos_step_ungapped(aligned_dict, start, stop, step, seqid):
+    """slicing an alignment should propogate the slice to aligned instances"""
+    aln = new_alignment.make_aligned_seqs(aligned_dict, moltype="dna")
+    sliced_aln = aln[start:stop:step]
     got = sliced_aln.seqs[seqid].seq
-    expect = expect.replace("-", "")
+    expect = aligned_dict[seqid][start:stop:step].replace("-", "")
+    assert got == expect
+
+
+@pytest.mark.parametrize("start", range(6))
+@pytest.mark.parametrize("stop", range(6))
+@pytest.mark.parametrize("step", range(-3, 0))
+@pytest.mark.parametrize("seqid", ("seq1", "seq2", "seq3", "seq4"))
+def test_alignment_slice_neg_step_gapped(aligned_dict, start, stop, step, seqid):
+    """slicing an alignment should propogate the slice to aligned instances"""
```

Comment:
don't these tests always create an empty record? That is, the following is True
```python
assert "ACGGTG"[1:3:-3] == "
```
You would need to reverse your ranges in the fixtures I think

Reply:
I'm not sure that reversing the range of the fixture will change anything. Pytest will do the combination of all fixtures. You're right that about 2/3rds of the combinations will result in an empty slice - but I thought it was worth checking that in both cases that were valid slices and not valid slices, the result was the same.

Reply:
Gotcha, I did not fully think this through. As all permutations of start / stop are being produced, sometimes start > stop.